include/dfu/suit_dfu_fetch_source.h

```diff
@@ -16,45 +29,39 @@
- *        Multiple such callbacks can be registered. The callbacks are called from inside
- *        the SUIT processor until one of them succeeds.
- *        Iteration will be broken if one of registered fetch sources starts to call
- *        @ref suit_dfu_fetch_source_write_fetched_data or @ref suit_dfu_fetch_source_seek.
  *
- *        The role of the callback is to fetch data from the provided URI and pass it to SUIT
- *        via @ref suit_dfu_fetch_source_write_fetched_data - the data does not have to be passed
- *        all at once, @ref suit_dfu_fetch_source_write_fetched_data can be called multiple times
- *        as new chunks of data arrive.
- *        If the data needs to be written at some offset (e.g. if the chunks arrive out of order)
- *        @ref suit_dfu_fetch_source_seek can be used to move the write pointer to the given offset.
+ * @details
+ * Fetch source callbacks are registered via suit_dfu_fetch_source_register.
+ * Multiple such callbacks can be registered. The callbacks are called from inside
+ * the SUIT processor until one of them succeeds.
+ * Iteration will be broken if one of registered fetch sources starts to call
+ * @ref suit_dfu_fetch_source_write_fetched_data or @ref suit_dfu_fetch_source_seek.
  *
- *        It is up to the implementer of this callback how the URI is resolved and how
- *        the data is fetched.
- *        If the function fails to fetch the data from the given URI it must return a negative
- *        error code.
+ * The role of the callback is to fetch data from the provided URI and pass it to SUIT
+ * via @ref suit_dfu_fetch_source_write_fetched_data - the data does not have to be passed
+ * all at once, @ref suit_dfu_fetch_source_write_fetched_data can be called multiple times
+ * as new chunks of data arrive.
+ * If the data needs to be written at some offset (e.g. if the chunks arrive out of order)
+ * @ref suit_dfu_fetch_source_seek can be used to move the write pointer to the given offset.
  *
- * @param[in]   uri URI from which the data should be fetched. Fetch source interprets it.
- *                  In case if fetch source is unable to retrieve requested resource, i.e. due to
- *                  unsupported protocol, it shall just fail, avoiding any prior call to
- *                  @ref suit_dfu_fetch_source_write_fetched_data or
- *                  @ref suit_dfu_fetch_source_seek
+ * It is up to the implementer of this callback how the URI is resolved and how
+ * the data is fetched.
+ * If the function fails to fetch the data from the given URI it must return a negative
+ * error code.
  *
- * @param[in]   uri_length Length of the uri, in bytes
+ * @param[in] uri         URI from which the data should be fetched. Fetch source interprets it.
+ *                        In case if fetch source is unable to retrieve requested resource,
+ *                        i.e. due to unsupported protocol, it shall just fail, avoiding any prior
```

Comment:
"Avoiding" sounds like a suggestion. I understand that's a hard requirement.

```suggestion
 *                        i.e. due to unsupported protocol, it shall just fail, without any
```